src/snabbdom.ts

```diff
@@ -138,15 +136,13 @@ export function init(modules: Array<Partial<Module>>, domApi?: DOMAPI) {
 
   function invokeDestroyHook(vnode: VNode) {
     let i: any, j: number, data = vnode.data;
-    if (data !== undefined) {
+    if (isDef(vnode.sel)) {
       if (isDef(i = data.hook) && isDef(i = i.destroy)) i(vnode);
       for (i = 0; i < cbs.destroy.length; ++i) cbs.destroy[i](vnode);
       if (vnode.children !== undefined) {
         for (j = 0; j < vnode.children.length; ++j) {
           i = vnode.children[j];
-          if (i != null && typeof i !== "string") {
-            invokeDestroyHook(i);
-          }
+          invokeDestroyHook(i);
```

Comment:
I don't think you can leave this `if` statement out. Children can potentially be strings or `undefined`.

Reply:
At least in TypeScript, [children is](https://github.com/snabbdom/snabbdom/blob/e203f711c60fde72fa391f0a8e927df4a376204b/src/vnode.ts#L16) `Array<VNode | string> | undefined`.

So yes, the condition here should be changed accordingly.

Is the reuse of `i` here really helping with performance? it makes it hard for the typing. Perhaps that is why it is `any`. But this code might have been converted to TypeScript, and not originally written with TypeScript types in mind. Would it be welcome to "explode" this reuse of `i`, so that we would have better types and so the IDE would actually help in development with this variable (preventing such an error as this)?